auth/auth.go

```diff
@@ -9,39 +9,39 @@ import (
 type Auth interface {
 	// Init the auth package
 	Init(opts ...Option) error
-	// Generate a new auth ServiceAccount
-	Generate(*ServiceAccount) (*ServiceAccount, error)
-	// Revoke an authorization ServiceAccount
+	// Generate a new auth Account
+	Generate(*Account) (*Account, error)
```

Comment:
I would also change this to Generate(id string, ...GenerateOption) and pass in Roles, Metadata, or whatever so that we prevent the Account Account stutter and have ability to extend.

Reply:
Example https://gist.github.com/asim/c149a22c1c433a9972cb72289368160c#file-auth-go-L3

Reply:
I agree, I'll make this change. Thanks